cannon/mipsevm/tests/evm_common_test.go

```diff
@@ -378,6 +378,95 @@ func TestEVMSingleStep_MovzMovn(t *testing.T) {
 
 }
 
+func TestEVMSingleStep_MfhiMflo(t *testing.T) {
+	var tracer *tracing.Hooks
+	versions := GetMipsVersionTestCases(t)
+	cases := []struct {
+		name  string
+		funct uint32
+	}{
+		{name: "mflo", funct: uint32(0x12)},
+		{name: "mfhi", funct: uint32(0x10)},
+	}
+	for _, v := range versions {
+		for i, tt := range cases {
+			testName := fmt.Sprintf("%v (%v)", tt.name, v.Name)
+			t.Run(testName, func(t *testing.T) {
+				var opt testutil.StateOption
+				val := Word(0xdeadbeef)
+				if tt.funct == 0x12 {
+					opt = testutil.WithLO(val)
+				} else {
+					opt = testutil.WithHI(val)
+				}
+				goVm := v.VMFactory(nil, os.Stdout, os.Stderr, testutil.CreateLogger(), testutil.WithRandomization(int64(i)), testutil.WithPC(0), testutil.WithNextPC(4), opt)
+				state := goVm.GetState()
+				rsReg := uint32(0)
+				rtReg := uint32(0)
+				rdReg := uint32(8)
+				insn := rsReg<<21 | rtReg<<16 | rdReg<<11 | tt.funct
```

Comment:
(nit) You can simplify this since for these instructions a lot of the bits are always zero:
```suggestion
				insn := rdReg<<11 | tt.funct
```